scripts/l2/l2-gen-addresses.sh

```diff
@@ -10,23 +10,45 @@ output=$($WALLETS_SCRIPT)
 # Path to the .env file
 ENV_FILE="$(pwd)/.env"
 
-# Process the output and update the .env file
-echo "$output" | while IFS= read -r line; do
-    if [[ $line == export\ GS_* ]]; then
-        # Extract variable name and value
-        var_name=$(echo "$line" | cut -d'=' -f1 | cut -d' ' -f2)
-        var_value=$(echo "$line" | cut -d'=' -f2-)
-        
-        # Update or add the variable in the .env file
-        if grep -q "^$var_name=" "$ENV_FILE"; then
-            sed -i.bak "s|^$var_name=.*|$var_name=$var_value|" "$ENV_FILE" && rm "${ENV_FILE}.bak"
-        else
-            echo "$var_name=$var_value" >> "$ENV_FILE"
-        fi
+# Check if addresses already exist in the .env file
+check_addresses_exist() {
+  for role in ADMIN BATCHER PROPOSER; do
+    if grep -q "^GS_${role}_ADDRESS=" "$ENV_FILE" || \
+        grep -q "^GS_${role}_PRIVATE_KEY=" "$ENV_FILE"; then
+      return 0
     fi
-done
+  done
+  return 1
+}
+
+if check_addresses_exist; then
+  echo "Error: Addresses or private keys already exist in .env file. Please remove them first."
```

Comment:
it might be inconvenient to require manual removing the keys. maybe add a flag to allow overwrite?

Reply:
how about:
- if existing keys exist, check if they are funded
- if funded, run teardown
- backup keys to another file, e.g. .env.bak-202411200953
- overwrite keys and continue script execution